engine/src/witness/common/cf_parameters.rs

```diff
@@ -21,19 +25,27 @@ pub type VersionedCcmCfParameters = VersionedCfParameters<CcmAdditionalData>;
 pub struct VaultSwapParameters {
 	pub refund_params: ChannelRefundParameters,
 	pub dca_params: Option<DcaParameters>,
-	pub boost_fee: Option<BasisPoints>,
-	pub broker_fees: Beneficiaries<AffiliateShortId>,
+	pub boost_fee: BasisPoints,
+	pub broker_fee: Beneficiary<AccountId>,
+	pub affiliate_fees: BoundedVec<AffiliateAndFee, ConstU32<MAX_AFFILIATES>>,
 }
 
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use cf_chains::MAX_CCM_ADDITIONAL_DATA_LENGTH;
+	use cf_chains::{ChannelRefundParameters, ForeignChainAddress, MAX_CCM_ADDITIONAL_DATA_LENGTH};
 
 	const MAX_VAULT_SWAP_PARAMETERS_LENGTH: u32 = 1_000;
 	const MAX_CF_PARAM_LENGTH: u32 =
 		MAX_CCM_ADDITIONAL_DATA_LENGTH + MAX_VAULT_SWAP_PARAMETERS_LENGTH;
 
+	const REFERENCE_EXPECTED_ENCODED: &[u8] = &[
```

Comment:
Maybe subjective, but I kind of like the way I broke down individual fields in `check_utxo_encoding` (https://github.com/chainflip-io/chainflip-backend/blob/main/state-chain/chains/src/btc/vault_swap_encoding.rs#L144) so we could see which bytes correspond to which field. Probably more important for BTC though where we were trying to compress the encoding.